File: Prácticas/P1/Ruby/civitas/lib/test_p1.rb
# encoding: utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative "dado.rb"
require_relative "mazo_sorpresas.rb"

module Civitas
  class TestP1
    NUM_JUGADORES = 4
    @@dado = Dado.instance
    @@mazo = MazoSorpresas.new
    def self.main
      @x=100
      @uno = 0
      @dos = 0
      @tres = 0
      @cuatro = 0
      for i in 1..@x do
        @valor = @@dado.quien_empieza(NUM_JUGADORES)
        case @valor
        when 1
          @uno = @uno + 1
        when 2
          @dos = @dos + 1
        when 3
          @tres = @tres + 1
        when 4
          @cuatro = @cuatro + 1 
        end
      end
      puts "\t\tValores de quien empieza\n"
      puts "Uno: " + @uno.to_s + "\n"
      puts "Dos: " + @dos.to_s + "\n"
      puts "Tres: " + @tres.to_s + "\n"
      puts "Cuatro: " + @cuatro.to_s + "\n"
      
      puts "\t\tValor de dado debug = off\n"
      @@dado.debug = false
      puts "Tirada uno: " + @@dado.tirar.to_s + "\n"
      puts "Tirada dos: " + @@dado.tirar.to_s + "\n"
      
      puts "\t\tValor de dado debug = on\n"
      @@dado.debug = true
      puts "Tirada uno: " + @@dado.tirar.to_s + "\n"
      puts "Tirada dos: " + @@dado.tirar.to_s + "\n"
      
      puts "\t\tUltimo resultado de DADO\n"
      @@dado.debug = false
      puts "Tirada uno: " + @@dado.tirar.to_s + "\n"
      puts "Resultado almacenado: " + @@dado.ultimo_resultado.to_s + "\n"
      puts "Tirada dos: " + @@dado.tirar.to_s + "\n"
      puts "Resultado almacenado: " + @@dado.ultimo_resultado.to_s + "\n"
      
      puts "\t\tSalgo de la carcel DADO\n"
      puts "Tirada uno: " + @@dado.salgo_de_la_carcel.to_s + "  " + @@dado.ultimo_resultado.to_s + "\n"
      puts "Tirada dos: " + @@dado.salgo_de_la_carcel.to_s + "  " + @@dado.ultimo_resultado.to_s + "\n"
      puts "Tirada tres: " + @@dado.salgo_de_la_carcel.to_s + "  " + @@dado.ultimo_resultado.to_s + "\n"
      
      @a = Sorpresa.new
      @b = Sorpresa.new
      @@mazo.al_mazo(@a)
      @@mazo.al_mazo(@b)
      
      @@mazo.siguiente
      
      @@mazo.inhabilitar_carta_especial(@b)
      @@mazo.habilitar_carta_especial(@b)
    end
  end
  TestP1.main
end

File: Prácticas/P1/Java/civitas/src/main/java/civitas/TestP1.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

/**
 *
 * @author Jose Padial
 */
public class TestP1 {

    /**
     * @param args the command line arguments
     */
    
    private static final int numJugadores = 4;
    private static Dado dado = Dado.getInstance();
    
    public static void main(String[] args) {
        int cero = 0, uno = 0, dos = 0, tres = 0, otro = 0;
        MazoSorpresas mazo = new MazoSorpresas();
        
        for(int i = 0; i < 100; ++i){
            int valor = dado.quienEmpieza(numJugadores);
            switch(valor){
                case 0: cero++; break;
                case 1: uno++; break;
                case 2: dos++; break;
                case 3: tres++; break;
                default: otro++; break;
            }
        }
        
        System.out.println("\t\tValores de quien empieza");
        System.out.println("Cero: " + cero);
        System.out.println("Uno: " + uno);
        System.out.println("Dos: " + dos);
        System.out.println("Tres: " + tres);
        System.out.println("Otro: " + otro);
        
        System.out.println("\t\tValor de dado debug = off");
        dado.setDebug(false);
        System.out.println("Tirada uno: " + dado.tirar());
        System.out.println("Tirada dos: " + dado.tirar());
        
        System.out.println("\t\tValor de dado debug = on");
        dado.setDebug(true);
        System.out.println("Tirada uno: " + dado.tirar());
        System.out.println("Tirada dos: " + dado.tirar());
        
        System.out.println("\t\tUltimo resultado de DADO");
        dado.setDebug(false);
        System.out.println("Tirada uno: " + dado.tirar());
        System.out.println("Resultado almacenado: " + dado.getUltimoResultado());
        System.out.println("Tirada dos: " + dado.tirar());
        System.out.println("Resultado almacenado: " + dado.getUltimoResultado());
        
        System.out.println("\t\tSalgo de la carcel DADO");
        System.out.println("Tirada uno: " + dado.salgoDeLaCarcel() + "  " + dado.getUltimoResultado());
        System.out.println("Tirada dos: " + dado.salgoDeLaCarcel() + "  " + dado.getUltimoResultado());
        System.out.println("Tirada tres: " + dado.salgoDeLaCarcel() + "  " + dado.getUltimoResultado());
        
        Sorpresa a = new Sorpresa();
        Sorpresa b = new Sorpresa();
        mazo.alMazo(a);
        mazo.alMazo(b);
        
        mazo.siguiente();
        
        mazo.inhabilitarCartaEspecial(b);
        mazo.habilitarCartaEspecial(b);
    }
    
}
